sklearn/tests/test_base.py

```diff
@@ -314,48 +312,131 @@ def transform(self, X, y=None):
     assert_equal(e.scalar_param, cloned_e.scalar_param)
 
 
-class TreeNoVersion(DecisionTreeClassifier):
-    def __getstate__(self):
-        return self.__dict__
+def test_pickle_version_warning_is_not_raised_with_matching_version():
+    iris = datasets.load_iris()
+    tree = DecisionTreeClassifier().fit(iris.data, iris.target)
+    tree_pickle = pickle.dumps(tree)
+    assert_true(b"version" in tree_pickle)
+    tree_restored = assert_no_warnings(pickle.loads, tree_pickle)
+
+    # test that we can predict with the restored decision tree classifier
+    score_of_original = tree.score(iris.data, iris.target)
+    score_of_restored = tree_restored.score(iris.data, iris.target)
+    assert_equal(score_of_original, score_of_restored)
 
 
 class TreeBadVersion(DecisionTreeClassifier):
     def __getstate__(self):
         return dict(self.__dict__.items(), _sklearn_version="something")
 
 
-def test_pickle_version_warning():
-    # check that warnings are raised when unpickling in a different version
-
-    # first, check no warning when in the same version:
+def test_pickle_version_warning_is_issued_upon_different_version():
     iris = datasets.load_iris()
-    tree = DecisionTreeClassifier().fit(iris.data, iris.target)
-    tree_pickle = pickle.dumps(tree)
-    assert_true(b"version" in tree_pickle)
-    assert_no_warnings(pickle.loads, tree_pickle)
-
-    # check that warning is raised on different version
     tree = TreeBadVersion().fit(iris.data, iris.target)
     tree_pickle_other = pickle.dumps(tree)
     message = ("Trying to unpickle estimator TreeBadVersion from "
-               "version {0} when using version {1}. This might lead to "
```

Comment:
Hope a35832cb60cc08e2552251ef1d2c7ee6e50b4b6e is as you intend it to be. I really didn't like the `replace` calls in the old tests. I assume you favour a global message template, two over repeating it in the tests. The other option would be to duplicate the template in the tests (which I think is probably the least optimal variant).